drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/enrollment.rb

```diff
@@ -365,6 +390,31 @@ def report_item_fields_from_enrollment(report_items:, enrollment:, report:) # ru
       report_item
     end
 
+    private def hp_targeting_criteria_complete(enrollment)
+      return false if enrollment.TimeToHousingLoss.blank? || enrollment.TimeToHousingLoss == 99 # V7.A
+      return false if enrollment.AnnualPercentAMI.blank? || enrollment.AnnualPercentAMI == 99 # V7.B
+      return false if enrollment.LiteralHomelessHistory.blank? || enrollment.LiteralHomelessHistory == 99 # V7.C
+      return false if enrollment.ClientLeaseholder.blank? || enrollment.ClientLeaseholder == 99 # V7.D
+      return false if enrollment.HOHLeaseholder.blank? || enrollment.HOHLeaseholder == 99 # V7.E
+      return false if enrollment.SubsidyAtRisk.blank? || enrollment.SubsidyAtRisk == 99 # V7.F
+      return false if enrollment.EvictionHistory.blank? || enrollment.EvictionHistory == 99 # V7.G
+      return false if enrollment.CriminalRecord.blank? || enrollment.CriminalRecord == 99 # V7.H
+      return false if enrollment.IncarceratedAdult.blank? || enrollment.IncarceratedAdult == 99 # V7.I
+      return false if enrollment.PrisonDischarge.blank? || enrollment.PrisonDischarge == 99 # V7.J
+      return false if enrollment.SexOffender.blank? || enrollment.SexOffender == 99 # V7.K
+      return false if enrollment.DisabledHoH.blank? || enrollment.DisabledHoH == 99 # V7.L
+      return false if enrollment.CurrentPregnant.blank? || enrollment.CurrentPregnant == 99 # V7.M
+      return false if enrollment.SingleParent.blank? || enrollment.SingleParent == 99 # V7.N
+      return false if enrollment.DependentUnder6.blank? || enrollment.DependentUnder6 == 99 # V7.O
+      return false if enrollment.HH5Plus.blank? || enrollment.HH5Plus == 99 # V7.P
+      return false if enrollment.CoCPrioritized.blank? || enrollment.CoCPrioritized == 99 # V7.Q
+      return false if enrollment.HPScreeningScore.blank? # V7.R
+      return false if enrollment.ThresholdScore.blank? # V7.S
```

Comment:
Where does this logic come from?  Can you add a comment pointing to the source doc on the HUD Exchange maybe?  Also, is there any utility in checking it with something like:
```
enrollment.TimeToHousingLoss.in?(known_values)
```
That would catch blanks, 99s, and any value that is irrelevant.  (Maybe we're doing that somewhere else?)

Reply:
I pulled the fields from V7 HP Targeting Criteria in the [FY 2024 HMIS Data Dictionary](https://files.hudexchange.info/resources/documents/HMIS-Data-Dictionary-2024.pdf), then mapped the fields to the enrollment data based on [drivers/hmis/lib/form_data/default/fragments/v7_hp_targeting_criteria.json](https://github.com/greenriver/hmis-warehouse/blob/ad2ee5ab286e771bea1746c90fd024e1698f1448/drivers/hmis/lib/form_data/default/fragments/v7_hp_targeting_criteria.json). The json file also includes the reference to the hud list with the relevant responses.

I'll add a comment around this in the method.

For the known_values, the values are all from the HudLists2024 data which includes the 99 values. I may be missing something, but to do this we would be recreating a bunch of the lists. We could possibly create a method that takes a relevant hud list as an argument and strips the '99' responses so we only get the valid codes.